### PR TITLE
added check to wait for all metrics data get collected and send to te…

### PR DESCRIPTION
**Pull request by edwardg8**

### CODE REVIEW

File: d2l-telemetry.js [hunk 1]
```diff
@@ -67,6 +72,27 @@
 			xhr.send(JSON.stringify(data));
 		},
 
+		checkAllMetricsCollected: function(perf) {
+			var check = false;
+			if (perf) {
+				var keys = Object.keys(perf);
+				var expectedKeys = ['d2l.page.load', 'd2l.page.visible',
```
Comment (edwardg8) [inline d2l-telemetry.js]: Definitely do not like hardcoding all the expected metrics, and looking for suggestions.

Comment (dlockhart) [inline d2l-telemetry.js]: Wouldn't this be better suited in the LMS code where it knows the names of the events it's waiting for?

Comment (edwardg8) [inline d2l-telemetry.js]: so we want to pass it from LMS to telemetry.js?

Comment (dlockhart) [inline d2l-telemetry.js]: We could do that, or just not call this method from the LMS until it has seen all the events it expects -- either way would work.

**Pull request merged**